app/main.cpp

```diff
@@ -485,6 +485,13 @@ int main( int argc, char *argv[] )
   QObject::connect( &activeProject, &ActiveProject::projectWillBeReloaded, &inputProjUtils, &InputProjUtils::resetHandlers );
   QObject::connect( &pw, &ProjectWizard::notify, &iu, &InputUtils::showNotificationRequested );
   QObject::connect( &iosUtils, &IosUtils::showToast, &iu, &InputUtils::showNotificationRequested );
+  QObject::connect( &syncManager, &SynchronizationManager::syncFinished, &activeProject, [&activeProject]( const QString & projectFullName, bool successfully, int version, bool reloadNeeded )
+  {
+    if ( reloadNeeded )
+    {
+      activeProject.reloadProject( activeProject.qgsProject()->homePath() );
```

Comment:
As we discussed, let's have a dedicated function for reloading of project (it can actually be this one when you remove the duplicated reload connection here https://github.com/lutraconsulting/input/pull/2047/files#diff-c1b709b8a90bef4573ddad8a9b2fbc5d5dbc85e4dc759f95b6d9905931fec24fR482). It does not need to receive any arguments as `activeProject` already has the correct `qgsProject` set.